zcu/known_keys.py

```diff
@@ -48,16 +53,19 @@ def mac_to_str(mac):
 
     return "%02x:%02x:%02x:%02x:%02x:%02x" % (mac[0], mac[1], mac[2], mac[3], mac[4], mac[5])
 
-def tagparams_keygen(params, key_prefix='Mcd5c46e', iv_prefix='G21b667b'):
+def tagparams_keygen(params, key_prefix='Mcd5c46e', iv_prefix='G21b667b', buggy=False):
```

Comment:
Can this be changed to pass in the hash function rather than the 'buggy' flag?
```py
def tagparams_keygen(params, key_prefix='Mcd5c46e', iv_prefix='G21b667b', hashfunc=None):
    # ...
    if hashfunc is None:
        hashfunc = ambda x: sha256(x).digest()
```